CMakeLists.txt

```diff
@@ -63,13 +41,6 @@ set (CMAKE_CXX_FLAGS_RELEASE "")
 set (CMAKE_CXX_FLAGS_DEBUG "")
 message(STATUS "CXX flags are: ${CMAKE_CXX_FLAGS}")
```

Comment:
I would really love if we can avoid overwriting all these standard flags and instead just use the defaults. 
Changes these can have unexpected consequences, e.g. a user might specify their own `CMAKE_CXX_FLAGS_RELEASE` which will have *zero* effect currently.